src/__tests__/use-app/use-app.test.tsx

```diff
@@ -102,6 +107,51 @@ describe('_app support', () => {
         expectDOMElementsToMatch(actual, expected);
       });
     });
+
+    describe('Custom props', () => {
+      it('are passed to App', async () => {
+        const { render } = await getPage({
+          nextRoot:
+            __dirname + '/__fixtures__/custom-app-with-gip-custom-props',
+          route: '/gip',
+        });
+        const { nextRoot: actual } = render();
+        const { container: expected } = renderWithinNextRoot(
+          <CustomAppWithGIPCustomProps
+            Component={CustomAppWithGIPCustomProps_GIP}
+            pageProps={{
+              fromPage: true,
+            }}
+            customProps={true}
+          />
+        );
+        expectDOMElementsToMatch(actual, expected);
+      });
+
+      it.each([
+        ['getServerSideProps', '/ssr', CustomAppWithGIPCustomProps_SSRPage],
+        ['getStaticProps', '/ssg', CustomAppWithGIPCustomProps_SSGPage],
+      ])('Page with %s', async (dataFetchingType, route, PageComponent) => {
+        const { render } = await getPage({
+          nextRoot:
+            __dirname + '/__fixtures__/custom-app-with-gip-custom-props',
+          route,
+        });
+        const { nextRoot: actual } = render();
+        const { container: expected } = renderWithinNextRoot(
+          <CustomAppWithGIPCustomProps
+            Component={PageComponent}
+            pageProps={{
+              fromPage: true,
+              propNameCollision: 'from-page',
+            }}
+            customProps={true}
+          />
+        );
+
+        expectDOMElementsToMatch(actual, expected);
+      });
+    });
```

Comment:
Is it necessary to test this feature with all available data fetching method? Or testing just one fetching method would provide the same test reliability?

Reply:
custom app props are passed to component in 3 different places for each fetching method so I think it is good to test all of it